src/scripts/components/widgets/modals/Modal.js

```diff
@@ -112,27 +112,13 @@ export const ModalScrollContent = styled.div`
 `
 
 class Modal extends Component<Props, void> {
-  renderModal () {
-    const { modalContent } = this.props
-    switch (modalContent) {
-      case MODAL.STAKE:
-        return <ModalStake />
-      case MODAL.SECURE:
-        return <ModalSecure />
-      case MODAL.GREAT_POWER:
-        return <ModalGreatPower />
-      case MODAL.CREATE_PASSWORD:
-        return <ModalCreatePassword />
-      case MODAL.ASK_PASSWORD:
-        return <ModalAskPassword />
-      case MODAL.SHOW_SEED:
-        return <ModalShowSeed />
-      case MODAL.PROFILE:
-        return <ModalProfile />
-      case MODAL.REWRITE_SEED:
-        return <ModalRewriteSeed />
-      case MODAL.RESTORE_ACCOUNT:
-        return <ModalRestoreAccount />
+  canClose = () => {
+    return this.props.modalContent !== MODAL.PROFILE
```

Comment:
You can no longer close the profile modal. The only way to dismiss it is to enter a username (and email if you want) and then continue to the next step.

Reply:
Yes that is still possible. I think to solve that we would have to rethink the flow a bit and reorder/combine the steps.

For now I think we want to leave the flow as is, plus most users would not do that anyway. But tagging @felipegaucho and @pedrocasa just to get their thoughts.

Reply:
ok @bent0b0x maybe merge this PR and the we refine the flow

Reply:
agree. I think that data integrity like this should not be an argument to change the flow here. It is always possible that the user stops halfway (e.g. his battery fails); what we should do is think of policies to recover from that. For example, for this case, we can use a default user name, or, on the next visit, ask the user to provide a username before doing anything else.